Source/ARTAuth.m

```diff
@@ -369,6 +380,20 @@ - (NSString *)getClientId {
     }
 }
 
+- (void)discardTimeOffset {
+    self.timeOffset = nil;
+}
+
+#pragma mark - Helper
+
+- (NSDate*)timeStampDateWithDateNow:(NSDate*)nowDate {
+    long long appNow = [nowDate timeIntervalSince1970]*1000;
+    long long seconds = ([self.timeOffset longLongValue] + appNow)/1000;
```

Comment:
You're losing precision here. `seconds` should be a `NSTimeInterval`, which has millisecond precision, while this just has second precision. If you just use `NSTimeInterval` you can get rid of all the `*1000` and `/1000` too.